src/org/zaproxy/zap/extension/stats/OptionsStatsPanel.java

```diff
@@ -0,0 +1,215 @@
+			statsdPanel.add(this.getStatsdHostField(), LayoutHelper.getGBC(1, 1, 1, 0.5));
+
+			statsdPanel.add(new JLabel(Constant.messages.getString("stats.options.label.statsd.port")), 
+					LayoutHelper.getGBC(0, 2, 1, 0.5));
+			statsdPanel.add(this.getStatsdPortField(), LayoutHelper.getGBC(1, 2, 1, 0.5));
+
+			statsdPanel.add(new JLabel(Constant.messages.getString("stats.options.label.statsd.prefix")), 
+					LayoutHelper.getGBC(0, 3, 1, 0.5));
+			statsdPanel.add(this.getStatsdPrefixField(), LayoutHelper.getGBC(1, 3, 1, 0.5));
+
+			panelMisc.add(statsdPanel, LayoutHelper.getGBC(0, 2, 1, 1.0D));
+
+			panelMisc.add(new JLabel(), LayoutHelper.getGBC(0, 10, 1, 0.5D, 1.0D));	// Spacer
+		}
+		return panelMisc;
+	}
+
+	private JCheckBox getInMemoryStatsEnabledField() {
+		if (inMemoryStatsEnabledField == null) {
+			inMemoryStatsEnabledField = new JCheckBox();
+			inMemoryStatsEnabledField.setText(Constant.messages.getString("stats.options.mem.enabled"));
+			inMemoryStatsEnabledField.setVerticalAlignment(javax.swing.SwingConstants.TOP);
+			inMemoryStatsEnabledField.setVerticalTextPosition(javax.swing.SwingConstants.TOP);
+		}
+		return inMemoryStatsEnabledField;
+	}
+
+	private JCheckBox getStatsdStatsEnabledField() {
+		if (statsdStatsEnabledField == null) {
+			statsdStatsEnabledField = new JCheckBox();
+			statsdStatsEnabledField.setText(Constant.messages.getString("stats.options.statsd.enabled"));
+			statsdStatsEnabledField.setVerticalAlignment(javax.swing.SwingConstants.TOP);
+			statsdStatsEnabledField.setVerticalTextPosition(javax.swing.SwingConstants.TOP);
+			statsdStatsEnabledField.addActionListener(new ActionListener() {
+
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					setStatsdFieldStates();
+				}});
+
+		}
+		return statsdStatsEnabledField;
+	}
+
+	private void setStatsdFieldStates() {
+		getStatsdHostField().setEnabled(statsdStatsEnabledField.isSelected());
+		getStatsdPortField().setEnabled(statsdStatsEnabledField.isSelected());
+		getStatsdPrefixField().setEnabled(statsdStatsEnabledField.isSelected());
+	}
+
+	private ZapTextField getStatsdHostField() {
+		if (statsdHostField == null) {
+			statsdHostField = new ZapTextField();
+		}
+		return statsdHostField;
+	}
+
+	private ZapPortNumberSpinner getStatsdPortField() {
+		if (statsdPortField == null) {
+			statsdPortField = new ZapPortNumberSpinner(0);
+		}
+		return statsdPortField;
+	}
+
+	private ZapTextField getStatsdPrefixField() {
+		if (statsdPrefixField == null) {
+			statsdPrefixField = new ZapTextField();
+		}
+		return statsdPrefixField;
+	}
+
+	@Override
+    public void initParam(Object obj) {
+	    OptionsParam optionsParam = (OptionsParam) obj;
+		StatsParam statsParam = optionsParam.getParamSet(StatsParam.class);
+
+	    getInMemoryStatsEnabledField().setSelected(statsParam.isInMemoryEnabled());
+	    getStatsdStatsEnabledField().setSelected(statsParam.isStatsdEnabled());
+	    getStatsdHostField().setText(statsParam.getStatsdHost());

```

Comment:
It should discard the undoable edits after setting the text (`discardAllEdits()`).